mindmeld/text_preparation/normalizers.py

```diff
@@ -195,28 +196,105 @@ def normalize(self, text):
         return unicodedata.normalize(self.normalization_type, text)
 
 
+class Lowercase(Normalizer):
+    """Lowercase Normalizer Class."""
+
+    def normalize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Normalized Text.
+        """
+        return text.lower()
+
+
+class RegexNormalizer(Normalizer):
+    """Normalization class that substitutes regex matches with either an empty string or a
+    replacement string."""
+
+    def __init__(self, regex_norm_rules):
+        """Creates a Regex Normalizer instance.
+
+        Args:
+            regex_norm_rules (List[Dict]): List of regex normalization rules represented as
+                dictionaries. The example rule below removes any text in parentheses.
+                {
+                    "description": "remove_text_in_parantheses",
+                    "pattern": "\(.+?\)",
+                    "replacement": ""
+                }
+        """
+        super().__init__()
+        self.regex_norm_rules = regex_norm_rules
+
+    def normalize(self, text):
```

Comment:
If you take the approach I described above, this method becomes a fair bit simpler:
```python
def normalize(self, text):
    return functools.reduce(lambda fn: fn(text), self.regex_norm_rules)
```